### PR TITLE
BugFix - Null Account Details Issue Fix

### PR DESCRIPTION
**Pull request by IamRanjeetSingh**
added null checks

Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced telemetry tracking with streamlined feature and log handling.
	- Introduced an optional duration parameter for feature usage tracking.

- **Bug Fixes**
	- Added error handling during log and feature record creation to improve stability.

- **Refactor**
	- Simplified request structures for better readability.
	- Updated methods for improved flexibility in telemetry data handling.
	- Streamlined the disposal process for the TelemetryQueueManager to enhance resource management.

### CODE REVIEW

File: Ginger/GingerCoreNET/Telemetry/TelemetryLogRecord.cs [hunk 1]
```diff
@@ -13,5 +13,7 @@ internal sealed class TelemetryLogRecord : TelemetryBaseRecord
         public required string Message { get; init; }
 
         public string Metadata { get; init; }
+
+        internal TelemetryLogRecord() { }
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Telemetry/TelemetryLogRecord.cs:17]: No, that is not possible

File: Ginger/GingerCoreNET/Telemetry/TelemetryFeatureRecord.cs [hunk 2]
```diff
@@ -13,5 +13,7 @@ internal sealed class TelemetryFeatureRecord : TelemetryBaseRecord
         public TimeSpan? Duration { get; init; }
 
         public string Metadata { get; init; }
+
+        internal TelemetryFeatureRecord() { }
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Telemetry/TelemetryFeatureRecord.cs:17]: No, that is not possible

File: Ginger/GingerCoreNET/Telemetry/TelemetryCollector.cs [hunk 3]
```diff
@@ -48,19 +48,25 @@ private AddLogsRequest CreateAddLogsRequest(IEnumerable<TelemetryLogRecord> logs
...
+                        CreationTimestamp = log.CreationTimestamp.ToString("O"),
+                        Level = log.Level ?? "",
+                        Message = log.Message ?? "",
+                        Metadata = log.Metadata ?? "",
+                    });
+                }
+                catch (Exception ex)
+                {
+                    _logger?.LogError("Error while creating {logRecord}\n{ex}", nameof(LogRecord), ex);
+                }
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Telemetry/TelemetryCollector.cs:69]: those validations are handled in higher level of abstraction. At this point of the code, there is no need to perform additional checks, and must send the record as it is.

File: Ginger/GingerCoreNET/Telemetry/TelemetryCollector.cs [hunk 4]
```diff
@@ -92,19 +98,25 @@ private AddFeaturesRequest CreateAddFeaturesRequest(IEnumerable<TelemetryFeature
...
-                    AppVersion = feature.AppVersion,
-                    UserId = feature.UserId,
-                    CreationTimestamp = feature.CreationTimestamp.ToString("O"),
-                    FeatureId = feature.FeatureId,
-                    Duration = feature.Duration != null ? feature.Duration.Value.ToString() : "",
-                    Metadata = feature.Metadata,
-                });
-                request.FeatureCount++;
+                    _logger?.LogError("Error while creating {featureRecord}\n{ex}", nameof(FeatureRecord), ex);
+                }
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Telemetry/TelemetryCollector.cs:119]: those validations are handled in higher level of abstraction. At this point of the code, there is no need to perform additional checks, and must send the record as it is.

File: Ginger/GingerCoreNET/Telemetry/TelemetryQueueManager.cs [hunk 5]
```diff
@@ -108,63 +109,64 @@ public void AddLog(eLogLevel level, string msg, TelemetryMetadata metadata)
...
+                Duration = duration,
                 Metadata = metadata.ToJSON(),
-            });
-        }
-
-        public IFeatureTracker StartFeatureTracking(FeatureId featureId)
-        {
-            return new FeatureTracker(featureId, onStop: AddFeatureUsage);
-        }
+            };
```
[the remark this answers is not in the document]
Comment (IamRanjeetSingh) [reply Ginger/GingerCoreNET/Telemetry/TelemetryQueueManager.cs:153]: it is fine

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**